pkg/ipamd/datastore/data_store.go

```diff
@@ -438,6 +439,11 @@ func (ds *DataStore) ReadBackingStore(isv6Enabled bool) error {
 		}
 	}
 
+	// Some entries may have been purged during recovery, so write to backing store
+	if err := ds.writeBackingStoreUnsafe(); err != nil {
```

Comment:
Recovery is from backing store right? Do we need this again?

Reply:
We need to write here to purge any stale entries that were present. Otherwise, those stale entries would remain in the file, but not be loaded into memory (into datastore).